File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode1290ConvertBinaryNumberInALinkedListToInteger.cpp
#include "LeetCode1290ConvertBinaryNumberInALinkedListToInteger.h"

int LeetCode1290ConvertBinaryNumberInALinkedListToInteger::getDecimalValue(ListNode* head)
{
    int result{};
    auto node = head;
    while (node!=nullptr)
    {
        result|= node->val;
        result <<= 1;
        node = node->next;
    }
    return result>>1;
}

File: LeetCodeProblemsGo/LeetCode1290ConvertBinaryNumberInALinkedListToInteger/LeetCode1290ConvertBinaryNumberInALinkedListToInteger.go
package LeetCode1290ConvertBinaryNumberInALinkedListToInteger

type ListNode struct {
	Val  int
	Next *ListNode
}

func getDecimalValue(head *ListNode) int {
	result := 0
	node := head
	for node != nil {
		result = (result << 1) | node.Val
		node = node.Next
	}
	return result
}
